### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/autoclass_tutorial.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/autoclass_tutorial.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/autoclass_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
+# قم بتحميل المثيلات التي تم تدريبها مسبقًا باستخدام AutoClass
+لم ترغب في إنشاء محول معماري لمؤشر الترابط الخاص بك، فهناك العديد من محولات المعمارية المختلفة التي يمكنك الاختيار من بينها. كجزء من الفلسفة الأساسية لـ 🤗 Transformers لجعل المكتبة سهلة وبسيطة ومرنة، فإن فئة `AutoClass` تستدل تلقائيًا وتحمّل البنية الصحيحة من نقطة تفتيش معينة. تسمح لك طريقة `from_pretrained()` بتحميل نموذج مُدرب مسبقًا لأي بنية بسرعة حتى لا تضطر إلى تكريس الوقت والموارد لتدريب نموذج من الصفر. إن إنتاج هذا النوع من التعليمات البرمجية غير المعتمدة على نقاط التفتيش يعني أنه إذا نجح رمزك مع نقطة تفتيش واحدة، فسيتم تشغيله مع نقطة تفتيش أخرى - طالما تم تدريبه لمهمة مماثلة - حتى إذا كانت البنية مختلفة.
+
+تذكر أن البنية تشير إلى هيكل النموذج، ونقاط التفتيش هي الأوزان لبنية معينة. على سبيل المثال، [BERT](https://huggingface.co/google-bert/bert-base-uncased) هي بنية، في حين أن `google-bert/bert-base-uncased` هي نقطة تفتيش. "النموذج" هو مصطلح عام يمكن أن يعني إما البنية أو نقطة التفتيش.
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:4]:
```suggestion
تذكر أن البنية تشير إلى هيكل النموذج، والنسخ هي الأوزان لبنية معمارية معينة. على سبيل المثال، [BERT](https://huggingface.co/google-bert/bert-base-uncased) هي بنية معمارية، في حين أن `google-bert/bert-base-uncased` هي نسخة. "النموذج" هو مصطلح عام يمكن أن يعني إما البنية أو نالنسخة.
```

File: docs/source/ar/autoclass_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+في هذا البرنامج التعليمي، ستتعلم كيفية:
+
+* تحميل محدد مُدرب مسبقًا
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:8]:
```suggestion
* تحميل مُجزّئ الرموز مُدرب مسبقًا
```

File: docs/source/ar/autoclass_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+* تحميل معالج صور مُدرب مسبقًا
+* تحميل مستخرج ميزات مُدرب مسبقًا
+* تحميل معالج مُدرب مسبقًا
+* تحميل نموذج مُدرب مسبقًا
+* تحميل نموذج كعمود فقري
+
+## AutoTokenizer
+
+تبدأ كل مهمة NLP تقريبًا بمحدد. يقوم المحلل اللغوي بتحويل إدخالك إلى تنسيق يمكن للنموذج معالجته.
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:17]:
```suggestion
تبدأ كل مهمة NLP تقريبًا بمُجزّئ للرموز. يقوم المُجزّئ بتحويل النص إلى شكل يمكن للنموذج معالجته.
```

File: docs/source/ar/autoclass_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+قم بتحميل محلل باستخدام [`AutoTokenizer.from_pretrained`]:
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:19]:
```suggestion
قم بتحميل المُجزّئ باستخدام [`AutoTokenizer.from_pretrained`]:
```

File: docs/source/ar/autoclass_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+```py
+>>> sequence = "In a hole in the ground there lived a hobbit."
+>>> print(tokenizer(sequence))
+{'input_ids': [101, 1999, 1037, 4920, 1999, 1996, 2598, 2045, 2973, 1037, 7570, 10322, 4183, 1012, 102], 
+ 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
+ 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
+```
+
+## AutoImageProcessor
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:37]:
```suggestion
## معالج الصور التلقائي (AutoImageProcessor)
 
```

File: docs/source/ar/autoclass_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+>>> from transformers import AutoImageProcessor
+
+>>> image_processor = AutoImageProcessor.from_pretrained("google/vit-base-patch16-224")
+```
+
+## AutoBackbone
+
+<div style="text-align: center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/Swin%20Stages.png">
+    <figcaption class="mt-2 text-center text-sm text-gray-500">A Swin backbone with multiple stages for outputting a feature map.</figcaption>
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:51]:
```suggestion
    <figcaption class="mt-2 text-center text-sm text-gray-500">الصورة توضح مخطط مراحل نموذج Swin.</figcaption>
```

File: docs/source/ar/autoclass_tutorial.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+يسمح لك [`AutoBackbone`] باستخدام النماذج المُدربة مسبقًا كعمود فقري للحصول على خرائط ميزات من مراحل مختلفة من العمود الفقري. يجب عليك تحديد أحد المعلمات التالية في [`~PretrainedConfig.from_pretrained`]:
+
+* `out_indices` هو فهرس الطبقة التي تريد الحصول على خريطة الميزات منها
+* `out_features` هو اسم الطبقة التي تريد الحصول على خريطة الميزات منها
+
+يمكن استخدام هذه المعلمات بشكل متبادل، ولكن إذا كنت تستخدم كلاً منها، فتأكد من أنها متوائمة مع بعضها البعض! إذا لم تمرر أيًا من هذه المعلمات، فسيقوم العمود الفقري بإرجاع خريطة الميزات من الطبقة الأخيرة.
+<div style="text-align: center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/Swin%20Stage%201.png">
+    <figcaption class="mt-2 text-center text-sm text-gray-500">A feature map from the first stage of the backbone. The patch partition refers to the model stem.</figcaption>
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:62]:
```suggestion
    <figcaption class="mt-2 text-center text-sm text-gray-500">صورة توضح خريطة ميزات من المرحلة الأولى للعمود الفقري.</figcaption>
```

File: docs/source/ar/autoclass_tutorial.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+الآن يمكنك الوصول إلى كائن `feature_maps` من المرحلة الأولى من العمود الفقري:
+
+```py
+>>> list(feature_maps[0].shape)
+[1, 96, 56, 56]
+```
+
+## AutoFeatureExtractor
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:89]:
```suggestion
## مستخرج الميزات التلقائي (AutoFeatureExtractor)
```

File: docs/source/ar/autoclass_tutorial.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+```py
+>>> from transformers import AutoFeatureExtractor
+
+>>> feature_extractor = AutoFeatureExtractor.from_pretrained(
+...     "ehcalabres/wav2vec2-lg-xlsr-en-speech-emotion-recognition"
+... )
+```
+
+## AutoProcessor
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:103]:
```suggestion
## المعالج التلقائي (AutoProcessor)
```

File: docs/source/ar/autoclass_tutorial.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+تتطلب المهام متعددة الوسائط معالجًا يجمع بين نوعين من أدوات المعالجة المسبقة. على سبيل المثال، يتطلب نموذج [LayoutLMV2](model_doc/layoutlmv2) معالج صور لمعالجة الصور ومحلل لغوي لمعالجة النص؛ يجمع المعالج بين الاثنين.
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:105]:
```suggestion
تتطلب المهام متعددة الوسائط معالجًا يجمع بين نوعين من أدوات المعالجة المسبقة. على سبيل المثال، يتطلب نموذج [LayoutLMV2](model_doc/layoutlmv2) معالج صور لمعالجة الصور ومُجزّئ لمعالجة النص؛ يجمع المعالج كليهما.
```

File: docs/source/ar/autoclass_tutorial.md [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+قم بتحميل معالج باستخدام [`AutoProcessor.from_pretrained`]:
+
+```py
+>>> from transformers import AutoProcessor
+
+>>> processor = AutoProcessor.from_pretrained("microsoft/layoutlmv2-base-uncased")
+```
+
+## AutoModel
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:115]:
```suggestion
## النموذج التلقائي (AutoModel)
```

File: docs/source/ar/autoclass_tutorial.md [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+
+أعد استخدام نفس نقطة التفتيش لتحميل بنية لمهمة مختلفة:
+
+```py
+>>> from transformers import TFAutoModelForTokenClassification
+
+>>> model = TFAutoModelForTokenClassification.from_pretrained("distilbert/distilbert-base-uncased")
+```
+
+بشكل عام، نوصي باستخدام فئة `AutoTokenizer` وفئة `TFAutoModelFor` لتحميل مثيلات مُدربة مسبقًا من النماذج. سيساعدك هذا في تحميل البنية الصحيحة في كل مرة. في البرنامج التعليمي التالي، تعرف على كيفية استخدام المحلل اللغوي ومعالج الصور ومستخرج الميزات والمعالج الذي تم تحميله حديثًا لمعالجة مجموعة بيانات للضبط الدقيق.
```
Comment (abodacs) [inline docs/source/ar/autoclass_tutorial.md:163]:
```suggestion
بشكل عام، نوصي باستخدام فئة `AutoTokenizer` وفئة `TFAutoModelFor` لتحميل نسخ لنماذج مُدربة مسبقًا. سيساعدك هذا في تحميل البنية الصحيحة في كل مرة. في البرنامج التعليمي التالي، ستتعرف على كيفية استخدام المُجزّئ اللغوي ومعالج الصور ومستخرج الميزات والمعالج الذي تم تحميله حديثًا لمعالجة مجموعة بيانات للضبط الدقيق.
```

**Review by abodacs — CHANGES_REQUESTED** [review]
سﻻم عليكم
@AhmedAlmaghz 
شكرا لمجهودك

تم اضافة بعض المﻻحظات

**Comment by AhmedAlmaghz** [comment]
Thank you @abidlabs

**Review by abodacs — APPROVED** [review]
Ready to merge

**Pull request closed without merging**